src/phylum/ci/ci_base.py

```diff
@@ -136,9 +136,24 @@ def phylum_project(self) -> str:
         return self._phylum_project
 
     @property
-    def phylum_group(self) -> str:
-        """Get the Phylum group."""
-        return self._phylum_group
+    def phylum_group(self) -> Optional[str]:
+        """Get the Phylum group in use.
+
+        The Phylum group name can be specified as an option or contained in the `.phylum_project` file.
+        A group name provided as an input option will be preferred over an entry in the `.phylum_project` file.
+
+        Return `None` when the group name is not available.
+        """
+        group_name = None
+        if self._phylum_group:
+            group_name = self._phylum_group
+        # The `.phylum_project` file may not exist in the case where it didn't exist initially and then it
+        # was not created because the project (with optional group) provided as input options already existed.
+        elif self.phylum_project_file.exists():
+            yaml = YAML()
+            settings_dict = yaml.load(self.phylum_project_file.read_text(encoding="utf-8"))
+            group_name = settings_dict.get("group_name")
+        return group_name
```

Comment:
A few things feel really weird about this:

1. The `phylum_group` property is now the "effective" value (i.e., either from the command line or from the file) while the `phylum_project` property is not (i.e., it is `None` if nothing was supplied on the command line).
2. It's possible to supply the project on the command line and the group in the `.phylum_project` file. This also makes it impossible to override a group in the `.phylum_project` file if you explicitly want to use user-owned projects instead.
3. (This is minor) Checking `.exists()` on a file is very LBYL... This is Python.

To remedy these issues. I think you should just split into 2 cases: Values from the command line or values from the file. Don't allow mixing the two. To know which case you are in, you just have to check `self._phylum_project is None`

And you should make a similar change to `def phylum_project` above so that the two properties are both the "effective" value.

```suggestion
        if self._phylum_project:
            # Project/Group supplied on command-line
            return self._phylum_group
        else:
            # Project/Group supplied in `.phylum_project`
            yaml = YAML()
            settings_dict = yaml.load(self.phylum_project_file.read_text(encoding="utf-8"))
            return settings_dict.get("group_name")
```

Reply:
Good points. It did feel a bit like a half measure...likely due to rushing through to get this solution in place. On the bright side, this change basically makes the `.phylum_project` file optional.